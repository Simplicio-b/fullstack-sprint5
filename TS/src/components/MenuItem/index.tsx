import MenuLink from './styles';

function MenuItem({ label }: any) : JSX.Element {
	return (
		<li className="menu__item">
			<MenuLink href="#home">
				<span>{label}</span>
			</MenuLink>
		</li>
	);
}

export default MenuItem;
