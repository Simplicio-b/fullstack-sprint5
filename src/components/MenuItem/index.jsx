import MenuLink from './styles';

function MenuItem({ label }) {
	return (
		<li className="menu__item">
			<MenuLink href="#home">
				<span>{label}</span>
			</MenuLink>
		</li>
	);
}

export default MenuItem;
