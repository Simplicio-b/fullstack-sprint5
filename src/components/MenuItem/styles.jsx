import styled from 'styled-components';

const MenuLink = styled.a`
	font-size: 0.8125rem;
	font-weight: 600;
	color: #000;
	text-transform: uppercase;
	padding-bottom: 5px;
	vertical-align: baseline;

	&:hover {
		border-bottom: 2px solid #000;
	}
`;

export default MenuLink;