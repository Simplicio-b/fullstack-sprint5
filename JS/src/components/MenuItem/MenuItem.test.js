import { render, screen } from '@testing-library/react';
import MenuItem from './index';

describe('Testando component MenuItem', () => {
	it('Teste de renderizacao', () => {
		render(<MenuItem label="Novidades" />);

		expect(screen.getByText('Novidades')).toBeInTheDocument();
	});

	it('Teste de snapshot', () => {
		const { container } = render(<MenuItem label="Novidades" />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
