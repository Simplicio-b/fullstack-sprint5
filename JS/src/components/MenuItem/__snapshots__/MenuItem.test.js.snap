// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testando component MenuItem Teste de snapshot 1`] = `
<li
  class="menu__item"
>
  <a
    class="sc-bdnxRM xJzDs"
    href="#home"
  >
    <span>
      Novidades
    </span>
  </a>
</li>
`;
